### PR TITLE
Add ability to link and an existing Bitbucket project

### PR DESCRIPTION
**Pull request by donovanmuller**
When clicking the _Link existing Bitbucket project_ button you now get a menu displaying a list of Bitbucket projects that can be selected:

screen shot 2018-02-16 at 18 10 01

screen shot 2018-02-16 at 18 10 35

On selecting a Bitbucket project, the project will be linked to the Gluon project and the Bitbucket project will be configured the same way a new Bitbucket project would. I.e. permissions, hooks etc. are configured.

screen shot 2018-02-16 at 18 12 13

Related to https://github.com/absa-subatomic/gluon/issues/21 for the event payload changes required.

### CODE REVIEW

File: .nyc_output/13187.json [hunk 1]
```diff
@@ -0,0 +1 @@
...-expr","locations":[{"start":{"line":8,"column":8},"end":{"line":8,"column":33}},{"start":{"line":8,"column":33},"end":{"line":8,"column":33}}]},"2":{"line":13,"type":"if","locations":[{"start":{"line":13,"column":8},"end":{"line":13,"column":7}},{"start":{"line":13,"column":8},"end":{"line":13,"column":7}}]}}},"/Users/donovan/dev/absa/core/subatomic/quantum-mechanic/src/openshift/base/options/AbstractOption.ts":{"path":"/Users/donovan/dev/absa/core/subatomic/quantum-mechanic/src/openshift/base/options/AbstractOption.ts","s":{"1":1,"2":1,"3":1,"4":1,"5":1,"6":1,"7":1},"b":{},"f":{"1":1},"fnMap":{"1":{"name":"AbstractOption","line":5,"loc":{"start":{"line":5,"column":4},"end":{"line":5,"column":92}}}},"statementMap":{"1":{"start":{"line":3,"column":0},"end":{"line":9,"column":1}},"2":{"start":{"line":5,"column":4},"end":{"line":5,"column":96}},"3":{"start":{"line":5,"column":26},"end":{"line":5,"column":38}},"4":{"start":{"line":5,"column":50},"end":{"line":5,"column":63}},"5":{"start":{"line":5,"column":75},"end":{"line":5,"column":92}},"6":{"start":{"line":9,"column":0},"end":{"line":9,"column":1}},"7":{"start":{"line":3,"column":22},"end":{"line":3,"column":36}}},"branchMap":{}}}
```
Comment (kbristow) [inline .nyc_output/13187.json]: Should this be here?

File: src/gluon/application/CreateApplication.ts [hunk 2]
```diff
@@ -49,13 +49,13 @@ export class CreateApplication implements HandleCommand<HandlerResult> {
     public handle(ctx: HandlerContext): Promise<HandlerResult> {
 
         // get memberId for createdBy
-        return memberFromScreenName(ctx, this.screenName)
```
Comment (kbristow) [inline src/gluon/application/CreateApplication.ts]: We should maybe start differentiating between gluon team and member and slack members and teams in general. Can get really confusing what team or member we are talking about.

Comment (donovanmuller) [reply src/gluon/application/CreateApplication.ts]: Yes, we definitely should. I'm busy changing as I go.

File: .nyc_output/13187.json (hunk 1 above)
Comment (donovanmuller) [reply .nyc_output/13187.json]: Nope, it shouldn't. Will remove.

**Comment by donovanmuller** [comment]
@kbristow I didn't see the review comments and merged. I'll remove that `.nyc_output/13187.json` file and then just create a follow on PR for that.

**Comment by donovanmuller** [comment]
Resolved https://github.com/absa-subatomic/quantum-mechanic/pull/37#issuecomment-366309475 via 0ca9b85783b631d6fd3e1fd2612564126561c029

**Pull request merged**